### PR TITLE
support json MasterInfo in ZK nodes

### PR DESCRIPTION
**Pull request by jdef**
* required for recent versions of mesos that store JSON in master ZK info nodes
* I left in support for older versions of mesos

### CODE REVIEW

**Comment by jdef** [comment]
@spacejam PTAL

**Comment by spacejam** [comment]
@jdef cool, what versions of mesos has this been tested with?

**Comment by jdef** [comment]
the one on my dev box: 0.26.0-0.2.145.ubuntu1404

**Comment by jdef** [comment]
Do you have an older mesos cluster kicking around that you want to test
this on?

File: rpc/zk.go [hunk 1]
```diff
@@ -236,6 +239,83 @@ func ClearZKState(
...
+	if addr := info.GetAddress(); addr != nil {
+		host = addr.GetHostname()
+		if host == "" {
+			host = addr.GetIp()
+		}
+		port = int(addr.GetPort())
+	} else {
+		host = info.GetHostname()
+		if host == "" {
+			host = addr.GetIp()
```
Comment (jdef) [inline rpc/zk.go]: bug! get this from `info`

File: rpc/zk.go [hunk 2]
```diff
@@ -236,6 +241,99 @@ func ClearZKState(
...
+type decoder func([]byte, interface{}) error
+
+var infoCodecs = map[string]decoder{
+	"info_": func(b []byte, i interface{}) error {
+		return proto.Unmarshal(b, i.(proto.Message))
+	},
+	"json.info_": json.Unmarshal,
+}
+
+// for example: 001 -> {info_001,proto.Unmarshal} or 001 -> {json.info_001,json.Unmarshal}
```
Comment (jdef) [inline rpc/zk.go]: move this comment

**Pull request merged**